### ISSUE TITLE
Peacefull?

### ISSUE THREAD

**Issue by Hangsna** [jsorrell/CarpetSkyAdditions#47]
Is it possible to make this playable also in peacefull difficulty?

**Comment by jsorrell** [comment]
A peaceful version would need a ton of changes from the typical progression because Zombies don't spawn (no iron, carrots, potatoes), Zombie Villagers don't spawn (no villagers), Blazes don't spawn (no Eyes of Ender so no End) etc. Do you have any ideas for settings that would allow progression in peaceful (while still being in the spirit of the standard progression)?

**Comment by Hangsna** [comment]
Yes, a lot of changes would need to be done and its probably more to think about then i can scope. Perhaps can peacefull mobs be used to get some drops? Perhaps the pig can have a small chance to drop the food used for breeding it (potato, carrot/beet). Grass can be moved to lower layers in steps (without enderman/silktouch).
I was also thinking of placing two villagers in the nether to have that gated beyond progression to nether (that also mean an iron farm is gated beyond nether). Getting water can be done from wandering trader but getting lava is trickier. It can be duplicated with dripstone but needs a couldron. Relying to much on the wandering trader is not that great but if he sells a couldron that would be a balanced way (since that is an item thats not needed to be bought over and over, just one is enough). Same with a lava bucket and dripstone, its not that great to wait for trader but still its only one buy each. That way of trading however means the player have to get foxes to spawn with emeralds before getting villagers to trade with.
Getting to The End is also hard. Before 1.9 celrics sold eyes of ender but thats not a really fun way to do it. It would be better if some action in the nether can give blaze rods. Perhaps if some over world mob (for example horses) is brought to the nether and killed there it can drop blaze rods. That makes it a bit trickier and still "uses the nether" for blaze rods. Another (perhaps more logical) option is if striders could be made to drop blaze rods, they spawn in peacefull so the player can make a farm for that in the nether.

I think something like this could be done for peacefull mode but there is a lot to think thru.

(Sidenote: The only real way to get amounts of slime in peacefull using the panda sneeze mechanic is really in the spirit of this pack more then just building a slime farm in non-peacefull!)

**Comment by Hangsna** [comment]
I have been playing this on peaceful now for some months and i really like it, i made some small modifications to loot table (added potato/carrot for junk fishing, and blazerod/quarz/gold nugget for strider) and gave myself 2 villagers to start with. It will need changes but i have some ideas for a progression line:

Get a composter (could be "Friendly bone meal" advancement). Compost apple/saplings for bone meal, make wheat and sell wheat to farmer villager. Make a barrel (could be "buck-load of water and fish" advancement) and use emeralds to get bucket of fish and also a water bucket. Since you start with villagers its a bit of a head start but still cant do much since you cant get string. So getting sheep to get beds is whats needed. Also transporting grass to lowest y-level (needs spreading without endermen) is a lot more tricky in peaceful so the "getting low" advancement could also be replaced/added with "down with the grass" or so. After beds there can be ironfarm (could be "ironical golems" or something for getting iron). Also that opens up getting cats and that leads to string and opens up fishing. Carrots/potato/beet could be added for fishing and perhaps within specific bioms. Lava and dripstone is available (if set to true) from wandering trader so nether is then possible and so are also stone.

Ideas for missing things are solvable. I changed drops of strider to get blazerod/quarz/gold but that could be done i better way. Perhaps if a strider is killed inside boundingbox of fortress it could drop blazerod and have wither skull as rare drop. Gold could be from striders in a bastion bounding box. Also in vanilla peaceful there is piglins that generate with the bastion. Perhaps a custom way of doing that could be to bring a pig inside a bastion bounding box and give it a crossbow (could be something like "nether bacon for bartering" for advancement), or perhaps giving it rotten flesh (that you can get from a cat gift). In normal minecraft there is an advancement for getting a ghast to the overworld and killing it there, perhaps the other way around (getting a mob to nether and killing it there, my suggestion is a goat) could be what gives ghast tear. Falling anvil on soulsand/wither skull could be used for beacon ("hit the wither on the head before it spawn" could be advancement).


I would love to help with ideas but sadly i am not able to make the edits that is then needed to implement. I however think this could be an interesting challenge. Skyblock without just going for a mob farm is something different.

**Comment by jsorrell** [comment]
I like the idea. The regular SkyBlock playthrough allows everything to be obtainable that is obtainable in vanilla. A peaceful version should do the same thing, except with peaceful vanilla. However, getting to the end on peaceful is not typically possible in peaceful, yet in your version, getting there is possible.

As a peaceful player, what do you think should be obtainable in a peaceful playthrough of this mod that is not obtainable in vanilla? For example, blaze rods (which allow elytra, endstone, shulker boxes, chorus), skulls/heads, froglights, tridents, prismarine, ghast tears, etc.

**Comment by Hangsna** [comment]
Hm, that is an interesting perspective, i have not thought of that! I was mostly looking for this to "fill gaps" that peaceful cant give but i understand that its more in line with the scope of the project to just make it possible to do what you can do in vanilla peaceful. My dream would be to be able to use this pack also in a normal (not skyblock) peaceful world to be able to get all things also in peaceful but i understand thats perhaps not what this is about. I play normal peaceful on a set seed where the end portal is already filled in so that the end is a possibility. Perhaps already filled in portal could be a solution if you wanted to make this just being able to get all that is obtainable in peaceful vanilla. But what i really want is to be able to place peaceful and still getting all things.
